src/export/StructureDefinitionExporter.ts

```diff
@@ -780,24 +796,20 @@ export class StructureDefinitionExporter implements Fishable {
     }
     rule.items.forEach(item => {
       if (item.type) {
-        // there might be a |version appended to the type, so don't include it while fishing
-        const [typeWithoutVersion, version] = item.type.split('|', 2);
-        const extension = this.fishForFHIR(typeWithoutVersion, Type.Extension);
+        // there might be a |version appended to the type, so try to use the version but fall back
+        // to any version if necessary
+        const extension = fishForFHIRBestVersion(this, item.type, Type.Extension);
         if (extension == null) {
           logger.error(
             `Cannot create ${item.name} extension; unable to locate extension definition for: ${item.type}.`,
             rule.sourceInfo
           );
           return;
         }
-        if (version != null && extension.version != null && version != extension.version) {
-          logger.warn(
-            `The ${typeWithoutVersion} extension was specified with version ${version}, but SUSHI found version ${extension.version}`,
-            rule.sourceInfo
-          );
-        }
         try {
           let profileUrl = extension.url;
+          const [, ...versionPart] = item.type.split('|');
+          const version = versionPart.join('|') || null;
           if (version) {
             profileUrl += `|${version}`;
           }
```

Comment:
I think this is another case where a variant of my regex replace trick might work.
```
const profileUrl = item.type.replace(/^[^|]+/, extension.url);
```

Reply:
Part of [70e31fc](https://github.com/FHIR/sushi/pull/1275/commits/70e31fc1b4bdb5eaab6119f21d08e44e038c24db)